### PR TITLE
Improve jms1/jms2 receive handling

### PR DESCRIPTION
**Pull request by mar-kolya**
This PR fixes instrumentation of `receiveNoWait` calls in case there is no message to consume. It also instruments `wait(timeout)` calls.

### CODE REVIEW

File: dd-java-agent/instrumentation/jms-1/src/main/java/datadog/trace/instrumentation/jms1/JMS1MessageConsumerInstrumentation.java [hunk 1]
```diff
@@ -75,28 +76,39 @@ public static void stopSpan(
...
               .withTag(DDTags.SPAN_TYPE, DDSpanTypes.MESSAGE_CONSUMER)
               .withTag(Tags.COMPONENT.getKey(), "jms1")
               .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CONSUMER)
               .withTag("span.origin.type", consumer.getClass().getName())
-              .withStartTimestamp(TimeUnit.MILLISECONDS.toMicros(startTime))
-              .startActive(true);
+              .withStartTimestamp(TimeUnit.MILLISECONDS.toMicros(startTime));
+
+      if (message == null) {
+        spanBuilder = spanBuilder.withTag(DDTags.RESOURCE_NAME, "No message to consume");
```
Comment (realark) [inline dd-java-agent/instrumentation/jms-1/src/main/java/datadog/trace/instrumentation/jms1/JMS1MessageConsumerInstrumentation.java]: To me this name looks like an instrumentation error. What about something like `JMS Receive` as a fallback name instead? Do we have other fallback jms names to standardize around? What do you think @tylerbenson?

Comment (mar-kolya) [reply dd-java-agent/instrumentation/jms-1/src/main/java/datadog/trace/instrumentation/jms1/JMS1MessageConsumerInstrumentation.java]: It looks like JMS2 would actually have same problem in pretty much the same code, I'll look into fixing that later

File: dd-java-agent/instrumentation/jms-1/src/test/groovy/JMS1Test.groovy [hunk 2]
```diff
@@ -28,124 +37,52 @@ class JMS1Test extends AgentTestRunner {
...
-    consumerTags.size() == 6
+    receivedMessage.text == messageText
+    assertTraces(TEST_WRITER, 2) {
+      producerTrace(it, 0, jmsResourceName)
+      trace(1, 1) { // Consumer trace
+        span(0) {
+          childOf TEST_WRITER.firstTrace().get(2)
+          serviceName "jms"
+          operationName "jms.consume"
+          resourceName "JMS receive: consumed from $jmsResourceName"
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/jms-1/src/test/groovy/JMS1Test.groovy]: consume and produce resource names now seem inconsistent.  I think I'm ok with that, though I just wanted to bring that to your attention.

File: dd-java-agent/instrumentation/jms-1/src/main/java/datadog/trace/instrumentation/jms1/JMS1MessageConsumerInstrumentation.java [hunk 3]
```diff
@@ -73,30 +75,44 @@ public static long startSpan() {
...
               .withTag(Tags.COMPONENT.getKey(), "jms1")
               .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CONSUMER)
               .withTag("span.origin.type", consumer.getClass().getName())
-              .withStartTimestamp(TimeUnit.MILLISECONDS.toMicros(startTime))
-              .startActive(true);
+              .withStartTimestamp(TimeUnit.MILLISECONDS.toMicros(startTime));
+
+      String resourceNamePrefix = "JMS " + method.getName() + ": ";
+      if (message == null) {
+        spanBuilder = spanBuilder.withTag(DDTags.RESOURCE_NAME, resourceNamePrefix + "no message");
```
Comment (realark) [inline dd-java-agent/instrumentation/jms-1/src/main/java/datadog/trace/instrumentation/jms1/JMS1MessageConsumerInstrumentation.java]: Can we drop the `no message` part? Names that look like errors tend to raise support questions. I think the resourceNamePrefix is good enough.

**Comment by mar-kolya** [comment]
This should no longer be a `breaking-change` since resources that were visible before have their names unchanged.

**Pull request merged**